### PR TITLE
Makes dim when gray a noop on windows terminals.

### LINKED ISSUE #58: (Windows) chalk.gray.dim not visible
Only when highlight in cmder.

In cmd:

|Normal|Highlight|
|---|---|
|image|image|

In cmder:

|Normal|Highlight|
|---|---|
|image|image|

### PR DESCRIPTION
**Pull request by jbnicolai**
Fixes #58 

Could use a quick sanity check.

### CODE REVIEW

**Comment by Qix-** [comment]
lgtm. Shame windows has mediocre support.

File: index.js [hunk 1]
```diff
@@ -69,6 +69,15 @@ function applyStyle() {
 	var nestedStyles = this._styles;
 	var i = nestedStyles.length;
 
+	// Turns out that on Windows dimmed gray text becomes invisible in cmd.exe,
+	// see https://github.com/chalk/chalk/issues/58
+	// If we're on Windows and we're dealing with a gray color, temporarily make 'dim' a noop.
+	var originalDim = ansiStyles.dim.open;
+	if ((nestedStyles.indexOf('gray') != -1 || nestedStyles.indexOf('grey') != -1) &&
+			process.platform === 'win32' && !/^xterm/i.test(process.env.TERM)) {
```
Comment (sindresorhus) [inline index.js]: Dry this up into a top level variable.

This check should also be first, so the engine can skip the other tests.

File: index.js [hunk 2]
```diff
@@ -69,6 +69,15 @@ function applyStyle() {
 	var nestedStyles = this._styles;
 	var i = nestedStyles.length;
 
+	// Turns out that on Windows dimmed gray text becomes invisible in cmd.exe,
+	// see https://github.com/chalk/chalk/issues/58
+	// If we're on Windows and we're dealing with a gray color, temporarily make 'dim' a noop.
+	var originalDim = ansiStyles.dim.open;
+	if ((nestedStyles.indexOf('gray') != -1 || nestedStyles.indexOf('grey') != -1) &&
```
Comment (sindresorhus) [inline index.js]: `!=` => `!==`

File: index.js [hunk 3]
```diff
@@ -69,6 +69,15 @@ function applyStyle() {
...
+			process.platform === 'win32' && !/^xterm/i.test(process.env.TERM)) {
+		ansiStyles.dim.open = '';
```
Comment (sindresorhus) [inline index.js]: Should we reset the `.close` too so not to introduce a moot dim reset escape?

Comment (jbnicolai) [inline index.js]: Considered that. Don't really think it matters, decided to go for the simplest solution.

File: index.js (hunk 2 above)
Comment (jbnicolai) [inline index.js]: :+1:

File: index.js (hunk 1 above)
Comment (jbnicolai) [inline index.js]: Check.

**Pull request merged**